source/user/mcscf.rst

```diff
@@ -1,57 +1,299 @@
+  mycas.fcisolver.irrep_nelec = {"A1g": (2, 1), "A1u": (1, 1), "E1ux": (1, 1), "E1uy": (1, 0)}
+
+4) Use automated strategies (``avas`` and ``dmet_cas``) to pick an active space based on AO orbitals you're targeting.
+For more details, see :source:`examples/mcscf/43-avas.py` and :source:`examples/mcscf/43-dmet_cas.py`.
+
+.. code-block:: python
+
+  from pyscf.mcscf import avas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, orbs = avas.avas(mf, ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+
+  
+
+.. code-block:: python
+
+  from pyscf.mcscf import dmet_cas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, mo = dmet_cas.guess_cas(mf, mf.make_rdm1(), ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+  mycas.kernel(mo)
+
+
+Frozen Core MCSCF
+-----------------
+
+To reduce to computational expense of CASSCF calculations, users can "freeze" orbitals thereby excluding them from optimization.
+
+Users can specify a number to free the N lowest orbitals:
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 6, 8)
+  mycas.frozen = 2
+  mycas.kernel()
+
+
+Users can also specify a list of orbital indices (0-based).
+These may be occupied, virtual, or active orbitals.
+
+.. code-block:: python
+  mycas = mcscf.CASSCF(myhf, 6, 8)
+  mycas.frozen = [0,1,26,27]
+  mycas.kernel()
+
+See :source:`examples/mcscf/19-frozen_core.py` for a complete example.
+
+
+State-Averaged Calculations
+---------------------------
+
+When dealing with states that are close in energy, it can be helpful to perform state average calculations where the multireference orbitals are optimized for multiple states.
+The ``state_average_`` function (note the hanging underscore) is a member function of ``CASCI``/``CASSCF`` objects and takes the weights of the states as input.
+The weights can be any normalized and non-negative array of values.
+
+.. code-block:: python
+
+  n_states = 5
+  weights = np.ones(n_states)/n_states
+  mycas = mcscf.CASSCF(mf, 4, 4).state_average_(weights)
+
+See :source:`examples/mcscf/15-state_average.py` for a complete example.
+
+
+By default, only a single spin and/or point group symmetry is targeted, but it is possible to target a mixture of both:
+
+.. code-block:: python
+
+  weights = [.5, .5]
+  solver1 = fci.direct_spin1_symm.FCI(mol)
+  solver1.wfnsym= 'A1'
+  solver1.spin = 0
+  solver2 = fci.direct_spin1_symm.FCI(mol)
+  solver2.wfnsym= 'A2'
+  solver2.spin = 2
+
+  mycas = mcscf.CASSCF(mf, 4, 4)
+  mcscf.state_average_mix_(mycas, [solver1, solver2], weights)
+  mycas.kernel()
+
+See :source:`examples/mcscf/41-state_average.py` for a complete example.
+
+
+Job Control
+-----------
+
+Optimization Settings
+"""""""""""""""""""""
+
+For CASSCF calculations, users may want to modify several of the convergence tolerances such as the energy tolerance (``conv_tol``), the orbital gradient tolerance (``conv_tol_grad``), and the maximum number of MCSCF iteration (``max_cycle_macro``).
```

Comment:
Replace "energy tolerance" and "orbital gradient tolerance" with convergence thresholds for the energy and the orbital gradient